### ISSUE TITLE
[Bug?] http 403 forbidden

### ISSUE THREAD

**Issue by Smenon10** [toni-moreno/snmpcollector#185]
Hello,
I recently installed snmpcollector. It's working great, however today after a "reload configuration" on the web GUI that took a while. I started to randomly get logged out of the UI when clicking on various menus. 

installed from rpm on centos7
snmpcollector v0.6.6 (git: a018a9a ) built at [2017-04-08 05:56:36] 

`[root@xxx-xxx-xxx snmpcollector]# tail -f http_access.log 
[Macaron] 2017-04-20 21:14:25: Started POST /login for 1.1.1.1
[Macaron] 2017-04-20 21:14:25: Completed /login 200 OK in 548.289µs
[Macaron] 2017-04-20 21:14:25: Started GET /api/rt/agent/info/version/ for 1.1.1.1
[Macaron] 2017-04-20 21:14:25: Completed /api/rt/agent/info/version/ 200 OK in 255.518µs
[Macaron] 2017-04-20 21:14:25: Started GET /api/rt/device/info for 1.1.1.1
[Macaron] 2017-04-20 21:14:25: Completed /api/rt/device/info 200 OK in 647.115µs
[Macaron] 2017-04-20 21:14:31: Started GET /api/rt/device/info/msr-b4i-ods04 for 1.1.1.1
[Macaron] 2017-04-20 21:14:31: Completed /api/rt/device/info/msr-b4i-ods04 200 OK in 64.20178ms
[Macaron] 2017-04-20 21:14:51: Started GET /api/rt/device/info/msr-b4i-ods04 for 1.1.1.1
[Macaron] 2017-04-20 21:14:51: Completed /api/rt/device/info/msr-b4i-ods04 403 Forbidden in 1.708994ms`

Also, I cannot currently reload the configuration and it gives me a pop up error and then logs me out.

I tried restarting snmpcollector and influxd with no luck.

Any ideas?

**Comment by toni-moreno** [comment]
Can you change your config.toml general   option logLevel = "debug"  restart and show us the generated snmpcollector.log  please?

**Comment by Smenon10** [comment]
I have attached the log. From the looks of it, it looks like the process is crashing and restart itself quickly. I think the "config reload" I did never finished and it set itself into a loop of some sort.

[snmpcollector.zip](https://github.com/toni-moreno/snmpcollector/files/945693/snmpcollector.zip)

**Comment by Smenon10** [comment]
I just tried rebooting the server and still same behavior. It's very clear that the process is restarting itself as doing a "ps aux | grep snmpcollector" shows different PID every 20-30 seconds.

**Comment by toni-moreno** [comment]
This seems to be the place where it is crashing:
````
2017-04-20T23:12:40.048344+00:00 msr-infr-snmp01 snmpcollector: panic: interface conversion: interface {} is string, not float64
2017-04-20T23:12:40.048685+00:00 msr-infr-snmp01 snmpcollector: goroutine 312 [running]:
2017-04-20T23:12:40.049020+00:00 msr-infr-snmp01 snmpcollector: github.com/toni-moreno/snmpcollector/pkg/data/measurement.(*Measurement).GetInfluxPoint(0xc420230000, 0xc420018930, 0x0, 0x0, 0x1)
2017-04-20T23:12:40.049395+00:00 msr-infr-snmp01 snmpcollector: /home/developer/src/gospace/src/github.com/toni-moreno/snmpcollector/pkg/data/measurement/measurement.go:382 +0xdc6
2017-04-20T23:12:40.049695+00:00 msr-infr-snmp01 snmpcollector: github.com/toni-moreno/snmpcollector/pkg/agent/device.(*SnmpDevice).measConcurrentGatherAndSend.func1(0xc421736fc0, 0xc420049320, 0xc421736fb0, 0xc421736fb8, 0xc420230000)
2017-04-20T23:12:40.050015+00:00 msr-infr-snmp01 snmpcollector: /home/developer/src/gospace/src/github.com/toni-moreno/snmpcollector/pkg/agent/device/measgather.go:36 +0x195
2017-04-20T23:12:40.050415+00:00 msr-infr-snmp01 snmpcollector: created by github.com/toni-moreno/snmpcollector/pkg/agent/device.(*SnmpDevice).measConcurrentGatherAndSend
2017-04-20T23:12:40.050738+00:00 msr-infr-snmp01 snmpcollector: /home/developer/src/gospace/src/github.com/toni-moreno/snmpcollector/pkg/agent/device/measgather.go:42 +0x15d
````

I will review code in a few hours , thank you very much to report it.

**Comment by sbengo** [comment]
Hi @Smenon10 , we reviewed the code, and it seems that the error is produced by the metric defined:

ID | Base OID | fieldName | DataSrcType | isTag
--- | ----|------- | ---- | -------
ifxAlias | .1.3.6.1.2.1.31.1.1.1.18 | ifxAlias | OCTETSTRING | false

As ifxAlias is not set as a Tag (`isTag = false`), when we try to log the metric: we are expecting a `float` instead of a `string` as it seems that we didn't consider that case. We will try to fix this ASAP.

Personally, I recommend you to set all fields that can return a `string` as a Tag (you  can set it on the definition of the metric). It will allow you to use InfluxDB statements to filter or group by.

Sorry and thanks for reporting this!

**Comment by toni-moreno** [comment]
Hi @Smenon10 I've found the error in a invalid conversion . I have update git master branch  with a fix , but this fix won't be available in a package (RPM/DEB/docker) in a few days.

As @sbengo told you you can configure IfAlias as a Tag  , to bypass this error (Although it is highly recommended)  you can do it with this simple sql sentence

```sh
$sqlite3 conf/snmpcollector.db
sqlite> update snmp_metric_cfg set istag=1 where id='ifxAlias';
```
And after restart  the collector again.

You would be able to unset ifxAlias as tag is you wish when we have released a new rpm with version 0.7.0

**Comment by Smenon10** [comment]
Thank you, that did indeed fix the problem! I will open another bug report for issues in the WebUI.
